server/src/main/java/org/apache/druid/server/coordination/SegmentLoadDropHandler.java

```diff
@@ -326,9 +326,10 @@ private List<DataSegment> getBootstrapSegments()
 
     List<DataSegment> bootstrapSegments = new ArrayList<>();
 
-    try (final CloseableIterator<DataSegment> iterator =
-             FutureUtils.getUnchecked(coordinatorClient.fetchBootstrapSegments(), true)) {
-      bootstrapSegments = ImmutableList.copyOf(iterator);
+    try {
+      final BootstrapSegmentsInfo response =
+          FutureUtils.getUnchecked(coordinatorClient.fetchBootstrapSegments(), true);
```

Comment:
Since this is going to block startup, should there be a timeout?

Reply:
Yeah, this uses the CoordinatorClient injected [here](https://github.com/apache/druid/blob/master/server/src/main/java/org/apache/druid/rpc/guice/ServiceClientModule.java#L111-L115), which has a standard retry policy with a max of 6 retries (~3 seconds total wait time) on transient errors. So processes will come up after ~3 seconds if the failures are persistent.

I plan to add an optional fail close strategy for tasks in a follow up, which will likely need a different retry policy than this standard policy here when enabled. I can try adding some tests that verify the different behaviors like startup doesn't block indefinitely, etc.

Reply:
Sounds good. What about success scenarios, where a historical or task is busy loading a lot of segments and thus may not respond to liveness probes?

Reply:
There is currently no overall startup timeout. If this is an issue, particularly for tasks, we may want to adjust the task bootstrapper such that it can at least respond to liveness probes.